Software/app/basic_azurertos/main.c

```diff
@@ -0,0 +1,190 @@
+/** Copyright © 2021 The Things Industries B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file main.c
+ *
+ * @copyright Copyright (c) 2021 The Things Industries B.V.
+ *
+ */
+
+#include "tx_api.h"
+#include "app.h"
+
+static void SystemClock_Config(void);
+static void Error_Handler(void);
+
+static int32_t Queue_value = 100;
+
+/* Define the ThreadX object control blocks */
+TX_THREAD thread_1;
+TX_THREAD thread_2;
+TX_QUEUE queue_0;
+TX_BYTE_POOL byte_pool_0;
+
+/* Define thread prototypes */
+void thread_1_entry(ULONG thread_input);
+void thread_2_entry(ULONG thread_input);
+
+int main(int argc, char **argv)
+{
+    /* Reset of all peripherals, Initializes the Flash interface and the Systick */
+    HAL_Init();
+
+    /* Configure the system clock */
+    SystemClock_Config();
+
+#if (GNSE_TINY_TRACER_ENABLE)
+    /* Initialize Tracer/Logger */
+    GNSE_TRACER_INIT()
+#endif
+
+    /* Initialize LED */
+    GNSE_BSP_LED_Init(LED_BLUE);
+
+    /* Enter the ThreadX kernel  */
+    tx_kernel_enter();
+}
+
+/* Define what the initial system looks like  */
```

Comment:
Extremely nitpicky, but this comment (and some others in this file) end with two spaces instead of one before closing